#! /bin/sh /usr/share/dpatch/dpatch-run
## 04_check_snmp_labels.dpatch by Oskar Liljeblad <oskar@osk.mine.nu>
##
## DP: check_snmp: fix multiple labels. (http://bugs.debian.org/647020)

@DPATCH@
diff -urNad '--exclude=CVS' '--exclude=.svn' '--exclude=.git' '--exclude=.arch' '--exclude=.hg' '--exclude=_darcs' '--exclude=.bzr' nagios-plugins-1.4.15~/plugins/check_snmp.c nagios-plugins-1.4.15/plugins/check_snmp.c
--- nagios-plugins-1.4.15~/plugins/check_snmp.c	2010-07-27 22:47:16.000000000 +0200
+++ nagios-plugins-1.4.15/plugins/check_snmp.c	2012-02-24 01:08:16.000000000 +0100
@@ -753,7 +753,7 @@
 					if (labels == NULL)
 						die (STATE_UNKNOWN, _("Could not reallocate labels\n"));
 				}
-				labels++;
+				nlabels++;
 				ptr = thisarg (ptr);
 				if (strstr (ptr, "'") == ptr)
 					labels[nlabels - 1] = ptr + 1;
